In [1]:
import pandas as pd

df = pd.read_csv("../data/raw/UrbanNest_Retail_Dirty_Customer_Order_Dataset.csv")
df.head()

,customer_id,order_id,product_id,phone_number,customer_name,gender,age_group,city,region,customer_segment,...,discount_rate,order_revenue,delivery_fee,payment_method,sales_channel,order_status,delivery_status,delivery_days,customer_rating,return_flag
0,CUST00001,ORD000001,PRD0017,2348082692761,Customer 1,NaN,60+,Abuja,North Central,Occasional,...,0.20,40000.0,2500,Bank Transfer,Mobile App,cancelled,delivered,2.0,4.0,NO
1,CUST00002,ORD000002,PRD0004,2348059532120,Customer 2,Male,46-60,Kaduna,North West,NaN,...,0.05,76000.0,1000,Card,Instagram,Completed,Delayed,14.0,5.0,No
2,CUST00003,ORD000003,PRD0002,2348056395726,Customer 3,F,60+,Kaduna,North West,Regular,...,0.00,375000.0,0,Card,Mobile App,NaN,delivered,1.0,NaN,Yes
3,CUST00004,ORD000004,PRD0020,2348080821128,Customer 4,Female,46-60,Port Harcourt,South South,Regular,...,0.00,375000.0,1000,Cash on Delivery,Store,Returned,Delayed,5.0,4.0,No
4,CUST00005,ORD000005,PRD0021,2348036427770,Customer 5,MALE,60+,Port Harcourt,South South,Premium,...,0.20,14000.0,2500,Wallet,Store,Completed,delivered,10.0,3.0,No


In [2]:
print(df.shape)
df.columns.tolist()

(5000, 25)


['customer_id',
 'order_id',
 'product_id',
 'phone_number',
 'customer_name',
 'gender',
 'age_group',
 'city',
 'region',
 'customer_segment',
 'order_date',
 'product_category',
 'product_name',
 'quantity',
 'unit_price',
 'discount_rate',
 'order_revenue',
 'delivery_fee',
 'payment_method',
 'sales_channel',
 'order_status',
 'delivery_status',
 'delivery_days',
 'customer_rating',
 'return_flag']

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 25 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customer_id       5000 non-null   object 
 1   order_id          5000 non-null   object 
 2   product_id        5000 non-null   object 
 3   phone_number      5000 non-null   int64  
 4   customer_name     5000 non-null   object 
 5   gender            4020 non-null   object 
 6   age_group         4137 non-null   object 
 7   city              5000 non-null   object 
 8   region            5000 non-null   object 
 9   customer_segment  4615 non-null   object 
 10  order_date        5000 non-null   object 
 11  product_category  5000 non-null   object 
 12  product_name      5000 non-null   object 
 13  quantity          5000 non-null   int64  
 14  unit_price        5000 non-null   int64  
 15  discount_rate     5000 non-null   float64
 16  order_revenue     5000 non-null   float64


In [4]:
df.describe()


,phone_number,quantity,unit_price,discount_rate,order_revenue,delivery_fee,delivery_days,customer_rating
count,5.000000e+03,5000.000000,5000.000000,5000.000000,5000.000000,5000.000000,4582.000000,4495.000000
mean,2.348055e+12,3.012400,34086.700000,0.100490,90577.465000,1701.300000,7.436927,3.848943
std,2.624920e+07,1.419665,37737.145514,0.070159,117650.928602,1199.965122,4.085777,1.147008
min,2.348010e+12,1.000000,3500.000000,0.000000,2800.000000,0.000000,1.000000,1.000000
25%,2.348032e+12,2.000000,7500.000000,0.050000,16468.750000,1000.000000,4.000000,3.000000
50%,2.348055e+12,3.000000,15000.000000,0.100000,38250.000000,1500.000000,7.000000,4.000000
75%,2.348078e+12,4.000000,40000.000000,0.150000,114000.000000,2500.000000,11.000000,5.000000
max,2.348100e+12,5.000000,120000.000000,0.200000,600000.000000,3500.000000,14.000000,5.000000


Checking for missing values

In [5]:
df.isna().sum()

customer_id           0
order_id              0
product_id            0
phone_number          0
customer_name         0
gender              980
age_group           863
city                  0
region                0
customer_segment    385
order_date            0
product_category      0
product_name          0
quantity              0
unit_price            0
discount_rate         0
order_revenue         0
delivery_fee          0
payment_method      415
sales_channel       411
order_status        705
delivery_status     794
delivery_days       418
customer_rating     505
return_flag         985
dtype: int64

Checking for duplicates

In [6]:
df.duplicated().sum()

np.int64(0)

In [7]:
df["gender"].value_counts(dropna=False)

gender
Male      1201
Female    1128
NaN        980
M          328
F          309
female     241
male       186
 m         174
 f         161
FEMALE     155
MALE       137
Name: count, dtype: int64

## Data quality issues found

1. **Missing values** in 10 columns (e.g. gender 980, return_flag 985, order_status 705).
2. **Inconsistent spelling and case**, e.g. gender has Male / M / male / MALE / " m ".
3. **Extra spaces** around some text values.
4. **Wrong data type**: order_date is stored as text, not a date.
5. **No duplicate rows** were found.